pkg/controller/update_cluster.go

```diff
@@ -556,49 +557,68 @@ func (c *M3DBController) shrinkPlacementForSet(
 		return err
 	}
 
-	_, removeInst, err := c.findPodInstanceToRemove(cluster, pl, pods)
+	_, removeInst, err := c.findPodsAndInstancesToRemove(cluster, pl, pods, desiredInstanceCount)
 	if err != nil {
-		c.logger.Error("error finding pod to remove", zap.Error(err))
+		c.logger.Error("error finding pods to remove", zap.Error(err))
 		return err
 	}
 
-	c.logger.Info("removing pod from placement", zap.String("instance", removeInst.ID()))
-	return c.adminClient.placementClientForCluster(cluster).Remove([]string{removeInst.ID()})
+	if len(removeInst) == 0 {
+		c.logger.Info("nothing to remove, skipping remove call")
+		return nil
+	}
+
+	removeIds := make([]string, len(removeInst))
+	for idx, inst := range removeInst {
+		removeIds[idx] = inst.ID()
+	}
+	c.logger.Info("removing instances from placement",
+		zap.String("instances", strings.Join(removeIds, ",")))
+	return c.adminClient.placementClientForCluster(cluster).Remove(removeIds)
 }
 
-// findPodInstanceToRemove returns the pod (and associated placement instace)
+// findPodsAndInstancesToRemove returns pods (and associated placement instances)
 // with the highest ordinal number in the stateful set AND in the placement, so
-// that we remove from the placement the pod that will be deleted when the set
+// that we remove from the placement pods that will be deleted when the set
 // size is scaled down.
-func (c *M3DBController) findPodInstanceToRemove(
+func (c *M3DBController) findPodsAndInstancesToRemove(
 	cluster *myspec.M3DBCluster,
 	pl placement.Placement,
 	pods []*corev1.Pod,
-) (*corev1.Pod, placement.Instance, error) {
+	desiredInstanceCount int,
+) ([]*corev1.Pod, []placement.Instance, error) {
 	if len(pods) == 0 {
 		return nil, nil, errEmptyPodList
 	}
+	if desiredInstanceCount < 0 {
+		msg := fmt.Sprintf("desired instance count is negative: %d", desiredInstanceCount)
```

Comment:
Shouldn't we also validate against `desiredInstanceCount = 0`?

Reply:
I think 0 should be a valid optin. Like we can do with `kubectl set scale 0`

Reply:
How can it be valid in terms of placement - where will the shards be relocated to?

Reply:
Hmm yeah maybe it does not make sense after all.